File: MAPE/Monitor/monitor.py
import os
import time
import requests
from flask import Flask, jsonify

# Configuration via variables d’environnement
MONITOR_HOST = os.getenv("MONITOR_HOST", "10.0.0.20")  # IP sur laquelle écouter
MONITOR_PORT = int(os.getenv("MONITOR_PORT", "5000"))  # Port

app = Flask(__name__)

@app.route('/monitor', methods=['GET'])
def monitor():
    try:
        r = requests.get('http://10.0.0.2:8181/health', timeout=2)

        if r.status_code == 200:
            data = r.json()
            return jsonify({
                "status": "OK",
                "data": data
            })
        else:
            return jsonify({
                "status": "ERROR",
                "message": f"Status code: {r.status_code}"
            }), 500
        
    except Exception as e:
        return jsonify({
            "status": "ERROR",
            "message": str(e)
        }), 500
    
if __name__ == '__main__':
    print(f"Starting monitor on {MONITOR_HOST}:{MONITOR_PORT}")
    app.run(host=MONITOR_HOST, port=MONITOR_PORT)
    

File: MAPE/Monitor/monitor.js
const express = require('express');
const request = require('request');
const app = express();

const MONITOR_PORT = process.env.MONITOR_PORT || 5000;
const TARGET_URL = process.env.TARGET_URL || "http://10.0.0.2:8181/health";

app.get('/monitor', (req, res) => {
    request({ url: TARGET_URL, timeout: 2000 }, (err, response, body) => {
        if (!err && response.statusCode === 200) {
            res.send({
                status: "OK",
                data: JSON.parse(body)
            });
        } else {
            res.status(500).send({
                status: "ERROR",
                message: err ? err.message : `Status code: ${response.statusCode}`
            });
        }
    });
});

app.listen(MONITOR_PORT, () => {
    console.log(`✅ Monitor server running on port ${MONITOR_PORT}`);
});
